### DISCUSSION TITLE
Troubleshoot text box issue

### DISCUSSION THREAD

**Discussion by davitsio** [LinwoodDev/Butterfly#195] [Q&A] [answered]
Hello, everyone!  i just recently found this lovely note program, and I like everything about it so far, except that I am having significant trouble editing text boxes.  In fact, I had so much difficulty initially that I thought that is just not a feature of this program and I reached out to the developer to ask if it could be done.  He was very kind and replied with good instructions for editing them, by selecting the "hand" and then clicking on the textbox.  I was surprised because I had tried that before.

I figured out that on my phone I have to somehow locate the exact center of a text box and click on it usually 10-15 times before I hit the right sweet spot to bring up the text edit option.  So clearly something isn't working correctly for me.  I wonder if anyone has a workaround for this?  I am using a Galaxy s22 Ultra with Android 12, and I installed Linwood Butterfly from F-Droid.  My partner is also having the exact same trouble on her Galaxy s22 (base model) running Android 12.

Any help would be greatly appreciated!

**Comment by CodeDoctorDE** [comment, marked as answer]
Strange...
It has the same detection as in the desktop version.
Do you have a large document? Do you tried zooming in?
Can you give me instructions what I need to click and which zoom level I need to have?

**Reply by davitsio** [reply]
Thank you for your reply and interest in helping!  I did a little experimenting today: I have a second phone that I tried to use Linwood Butterfly notes on today (a Google Pixel 5a). I put my Galaxy s22 side-by-side with the Pixel and I installed the same version and did the same exact steps on both at the same time.  On the Pixel, I have no trouble at all selecting things with the hand....it works perfectly.  But on the Galaxys s22 the hand select barely works.  No matter what I have drawn on the scree or what I have typed in text boxes, I have to get lucky to succeed in selecting with the hand.

The zoom level doesn't seem to matter much, as it does the same whether I am zoomed in or out.  Being Zoomed in very close seems like it *might* make it easier to select things, but it still doesn't work most of the time, so I am not sure.  The documents are not large; I am just creating blank documents, putting in small text boxes and drawings to test with.

Because it works on my Pixel 5a perfectly, then it seems that either the issue is with the Galaxy phones in general, or specifically with something I have configured on my (and my partners) Galaxy phones?  Perhaps the best question to think about then, is there anything that I might have configured that would break screen digitizer selections in the program?  I have no issues drawing in Butterfly notes, so the  touch detection seems correct in general.

**Reply by CodeDoctorDE** [reply]
Strange...
It uses the same detection as drawing...
Does the general document context menu open? Because if you click on a blank spot it needs to be opened

**Reply by davitsio** [reply]
That's a good question.  You are correct that the general document context menu doesn't open usually either.  Sometimes it does, but it has the same problems as the hand select.  It's very interesting, because it seems like *only* the hand tool has issues for me; the pen, path eraser, text box creation, area select, label making ALL work perfectly where I tap and when I tap.  But the hand tool doesn't select usually, nor does it bring up the context menu on blank space usually.

**Reply by CodeDoctorDE** [reply]
Okay...
I can't replicate the bug (I have no galaxy). If you have a step-by-step guide how and when this happen, this would be great.

**Reply by CodeDoctorDE** [reply]
On my galaxy phone it works (with android 12)

**Reply by davitsio** [reply]
Yes,  replicating the problem is very easy for me to do.  Steps:
1. Open Linwood Butterfly Notes
2. Create new blank note
3. Click on draw, text, or path and make any shape or any text on the screen.
4. click on hand select tool.
5. Attempt to select the object you created.  It will not work.  

As I mentioned earlier, this happens on 2 of my Android 12 galaxy phones, and if you are running a recent Galaxy phone with Android 12, I feel that the logical explanation may be that my software is somehow different.  I always remove "bloatware" and other carrier packages when I can, and I have done that on these phones via ADB Android Tools.  Everything works for me, except hand select in this note program, but still it could be related.  Is there a particular package or library that is called by hand select that is not by the rest of the functions?  I am just trying to troubleshoot.

**Reply by davitsio** [reply]
I actually just realized that the canvas (background) of a blank note should be clickable too, with the hand tool.  That function is similarly not working.  So there doesn't even need to be an object for the hand tool to fail selecting it for me.  It *does* work about 15% of the time, and fails about 85% of the time.  Meaning, if I continually click on the screen, it will eventually after 8 or 9 taps finally select the item or bring up the hand menu.  It is random though, and sometimes selects successfully immediately or on the second tap.  But statistically, it works about 15% of the time.

**Reply by CodeDoctorDE** [reply]
Hmm, okay.
It's very difficult to fix it when it doesn't happen on my devices...
Can you try the web version?
